src/H5Tconv.c

```diff
@@ -2753,131 +2795,205 @@ H5T__conv_struct_opt(H5T_t *src, H5T_t *dst, H5T_cdata_t *cdata, const H5T_conv_
  *-------------------------------------------------------------------------
  */
 static herr_t
-H5T__conv_enum_init(H5T_t *src, H5T_t *dst, H5T_cdata_t *cdata)
```

Comment:
The enum conversion routine had to be rewritten slightly so that it makes a copy of the source and destination datatypes, caches them and works directly on those, rather than modifying the original datatypes when it sorts the enum members by name or value. Otherwise, removing the datatype copies higher up in the library would cause the datatype an object was created with to be surprisingly modified.

While this was rewritten mostly from a functionality perspective, some synthetic benchmarking of converting 100,000,000 enum elements shows very little change in conversion performance (usually in favor of this update), to the point that it's just noise.